README.md

```diff
@@ -96,6 +96,63 @@ end
 Passing `?filters[user_posts_on_date]=10/12/20` will call the `user_posts_on_date` scope with
 `10/12/20` as the the first argument, and will grab the `user_id` and `blog_id` out of the params and pass them as a hash, as the second argument.
 
+### Scope Arguments with Types
+
+Sometimes you need the argument to a scope to be parsed using Sift types. Scope argument types can be specified by passing a hash to the  `type:` keyword,
+as follows:
+
+```ruby
+class Post < ActiveRecord::Base
+  scope :with_priority, ->(int_array) { where(priority: int_array) }
+end
+
+class PostsController < ApplicationController
+  include Sift
+
+  filter_on :with_priority, type: { scope: :int }
+
+  def index
+    render json: filtrate(Post.all)
+  end
+end
+```
+Passing `?filters[with_priority]=[1,2]` will call the `with_priority` scope with the array, `[1, 2]`, instead of `"[1,2]"`
```

Comment:
There doesn't seem to be much consensus over how arrays should be used in query parameters.
https://medium.com/raml-api/arrays-in-query-params-33189628fa68

But `=1,2` appears to be a more common pattern than `=[1, 2]`

Reply:
Sift uses the JSON standard for array parameters. My(unpopular?) opinion is that we should use either the Rails standard `filters[foo][]=1&filters[foo][]=2` or POST JSON and use the `X-http-method-override` header to specify GET simulation.

Reply:
I do think this PR is not affected by that decision, however. We could change this code to parse however we want:

https://github.com/procore/sift/blob/master/lib/sift/value_parser.rb#L24-L33

Reply:
Not a fan of `filters[foo][]=1&filters[foo][]=2` is that really a Rails standard? Can you explain `X-http-method-override` with GET simulation in more detail?

Reply:
By setting a header with key `X-http-method-override` and value `GET` in a JSON POST request (for example) you can specify that the request should be handled as a GET in servers that support this header (Rails servers).

So you can POST with a body of `{"filters": {"foo": [1,2]}}` with rails treating this as a GET with params of `{ filters: {foo: [1,2]}}`. This sidesteps all issues with URL length, and allows your request to use actual JSON instead of mixing URL parameters with JSON. In practice it works really well. Here is example code from the vector_processing repo:

```
      def large_get(path, params, timeout: 180)
        url = vapid_uri_base + path
        retries = 0
        begin
          raw_response = RestClient::Request.execute(
            url: url,
            method: :post,
            headers: {'Accept' => 'application/json;charset=utf-8', 'Content-Type' => 'application/json', 'Authorization' => authorization.authorization_header, 'X-Http-Method-Override' => 'get'},
            open_timeout: 3,
            timeout: timeout,
            verify_ssl: VectorProcessing::Vapid::Authorization.verify_ssl?,
            payload: params.to_json
          )
          JSON.parse(raw_response.body.force_encoding('UTF-8'))
        rescue RestClient::Unauthorized
          retry if authorization.refresh_success? && (retries+=1) < 5
        rescue RestClient::ExceptionWithResponse => rcewr
          raise ErrorResponseException.new("Error Response #{rcewr.response&.code || 'missing'} to X-Http-Method-Override 'get' on POST '#{url}'")
        end
      end
```
This requires no change to Vapid; it's supported by Rails out-of-the-box.